src/nnet3/nnet-test-utils.cc

```diff
@@ -804,6 +804,24 @@ static void GenerateRandomComponentConfig(std::string *component_type,
       os << "input-dim=" << input_dim << " output-dim=" << output_dim;
       break;
     }
+    case 17: {
+      *component_type = "Convolutional1dComponent";
+      int32 patch_stride = 10 + Rand() % 50, patch_step = 1 + Rand() % 4, 
+	    patch_dim = 4 + Rand () % 5;
+
+      // decrease patch_stride until (patch_stride - patch_dim) % patch_step == 0
+      while ((patch_stride - patch_dim) % patch_step != 0)
```

Comment:
In nnet2, in_deriv is supposed to be resized by the Backprop routine, that
it why it is typed CuMatrix not CuMatrixBase.  In nnet2, it's different.


On Sun, Sep 20, 2015 at 5:34 PM, Yiming Wang <notifications@github.com>
wrote:

> In src/nnet3/nnet-test-utils.cc
> <https://github.com/kaldi-asr/kaldi/pull/156#discussion_r39935031>:
>
> > @@ -804,6 +804,24 @@ static void GenerateRandomComponentConfig(std::string *component_type,
> >        os << "input-dim=" << input_dim << " output-dim=" << output_dim;
> >        break;
> >      }
> > +    case 17: {
> > +      *component_type = "Convolutional1dComponent";
> > +      int32 patch_stride = 10 + Rand() % 50, patch_step = 1 + Rand() % 4,
> > +	    patch_dim = 4 + Rand () % 5;
> > +
> > +      // decrease patch_stride until (patch_stride - patch_dim) % patch_step == 0
> > +      while ((patch_stride - patch_dim) % patch_step != 0)
>
> Sorry. no while needed. will replace it with a single expression.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/156/files#r39935031>.
>